marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/extensions/runFlow.sjs

```diff
@@ -39,8 +39,8 @@ function post(context, params, input) {
       uris = datahub.hubUtils.normalizeToArray(params.uri || options.uris);
       query = cts.documentQuery(uris);
     } else {
-      let identifier = options.identifier || flow.identifier;
-      query = identifier ? cts.query(identifier) : null;
+      let sourceQuery = options.sourceQuery || flow.sourceQuery;
+      query = sourceQuery ? cts.query(sourceQuery) : null;
     }
```

Comment:
@aebadirad ,
Not sure if the logic here has to be changed, I have just renamed identifier to sourceQuery

Reply:
we'll have to pick it off flow.options and make it a bit more complex than a ternary operator.

Also have to pull off the step declaration too. So yes, please change the logic.

Reply:
Make another PR for it, since this is now required to run the flow